subsys/random/rand32_timer.c

```diff
@@ -14,21 +14,38 @@
  * provide a random number generator.
  */
 
```

Comment:
@decsny, pull over the attribution information from the xoshiro code into this header